### DISCUSSION TITLE
Deploy TM1py into Google Cloud Function?

### DISCUSSION THREAD

**Discussion by macsir** [cubewise-code/tm1py#503] [Ideas]
Hi, @MariusWirtz 
Just want to know if tm1py can be deployed as google cloud function to be run to load data from BigQuery table into TM1 locally? I guess not. But could you advise if it is possible in other ways? Thanks.

**Comment by MariusWirtz** [comment]
Hi @macsir,

If you run TM1 locally, why not just run python locally as well and query data from Big Query with py?
Once you have data in py you can easily move it to a TM1 cube with TM1py.

I did a project recently to read/write to google sheets and we ran python locally on their TM1 server.
It worked great to connect to the google systems through a google python library. I assume it would work similarly for Big Query.

**Comment by macsir** [comment]
Thanks. The thing is if all things are in cloud, we can easily use pub/sub to know when the source table has been updated so I can trigger TM1 to load accordingly and neatly. However, if using local TM1py from the sever, I have to depend on some manual email notification to trigger which breaks the automation. IT is still working on some network configuration to allow for cloud talking to TM1 sever. Hopefully I can use user/password in TM1py in cloud to start the process.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by MariusWirtz** [comment]
I see. If all things are in the cloud that makes sense. If google cloud offers to run python, I would be surprised if TM1py doesn't work. TM1py has no eccentric dependencies except maybe for pandas (which is an optional dependency anyways).

If you do run py locally, you can also run it as a windows service, that checks every minute or so if there is "work" to do. That could make a neat architecture as well.

If python runs in the cloud, you would have to think about how you expose the TM1 REST API to the internet, which is not entirely unproblematic.

**Reply by macsir** [reply]
Forgot the reply. Yes, our IT has enabled it and using TM1py with username/password works fine. Thanks.
